workflows_core/utils/document_list.py

```diff
@@ -0,0 +1,69 @@
+from collections import UserList
+from hashlib import new
+from typing import Any, Dict, List, Union
+
+from workflows_core.utils.document import Document
+
+
+class DocumentList(UserList):
+    data: List[Document]
+
+    def __init__(self, initlist=None):
+        if initlist is not None:
+            for index, document in enumerate(initlist):
+                if not isinstance(document, Document):
+                    initlist[index] = Document(document)
+        super().__init__(initlist)
+
+    def __repr__(self):
+        return repr(self.data)
+
+    def __getitem__(self, key: Union[str, int]) -> Union[Document, "DocumentList"]:
+        if isinstance(key, str):
+            return [document[key] for document in self.data]
+        elif isinstance(key, slice):
+            return self.__class__(self.data[key])
+        elif isinstance(key, int):
+            return self.data[key]
+
+    def __setitem__(self, key: Union[str, int], value: Union[Any, List[Any]]):
+        if isinstance(key, str):
+            if isinstance(value, list):
+                for document, value in zip(self.data, value):
+                    document[key] = value
+            else:
+                for document in self.data:
+                    document[key] = value
+        elif isinstance(key, int):
+            self.data[key] = value
+
+    def to_json(self):
+        return [document.to_json() for document in self.data]
+
+    def remove_tag(self, field: str, tag: str, label_field: str = "label") -> None:
```

Comment:
I think these tag operations will need to be re-written

Reply:
Few developer use-cases I've come across:
- If a user supplies the `field` but it includes the `label_field` how does it handle that? 
- what if the `field` is missing? The appropriate error treatment is therefore to provide an empty list right? I don't think that's the behavior here

Reply:
input = `_keyphrase_.text` -> list -> `[{"label": "yes"}. {"label": "no"}]`
label_field = "label"
tag = "yes"
output = `[{"label": "no"}]`